### PR TITLE
86cujnfuc/trip listing api integration

### PR DESCRIPTION
**Pull request by akshay-krishna**
## Overview
This pull request introduces the integration of API in the trip listing page.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,9 @@
 -   [86cugqgpt] Implemented filters and sort for action bar
 -   [86cueubmz] Implemented trip booking page and booking details form UI
 -   # [86cugexhf] Landing page illustration, action bar incorporated and routing to trip listing page.
+-   # [86cugexhf] Landing page illustration, action bar incorporated and routing to trip listing page.
 -   [86cukqvct] Implemented API layer
 -   [86cumak5j] Enhanced API layer and added converters for requests and responses
 -   [86cueujxt] Implemented bus search functionality for action bar and action bar filters, handling query params, corresponding reducer busSearchSlice and toggle animation
+-   [86cujnfuc] API integration of Trip Listing Page.
+-   [86cueujxt] Implemented bus search functionality for action bar and action bar filters, handling query params, corresponding reducer busSearchSlice and toggle animation
```
Comment (abhijith-biju) [inline CHANGELOG.md]: Remove duplicate lines.

File: src/components/FullScreenLoader/FullScreenLoader.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import Backdrop from '@mui/material/Backdrop';
+import CircularProgress from '@mui/material/CircularProgress';
+
+interface IFullScreenLoaderProps {
+    open: boolean;
+}
+
+export default function FullScreenLoader({ open }: IFullScreenLoaderProps) {
```
Comment (abhijith-biju) [inline src/components/FullScreenLoader/FullScreenLoader.tsx]: I think using arrow functions would be better for uniformity.

File: src/components/loadMore/LoadMore.tsx [hunk 3]
```diff
@@ -1,18 +1,50 @@
...
 
-const LoadMore = () => {
+const LoadMore = ({
+    resultLength,
+    page,
+    setPage,
+    btnLoading,
+    setBtnLoading,
+}: {
+    resultLength: number;
```
Comment (RJR-003) [inline src/components/loadMore/LoadMore.tsx]: Could write the types seperately

**Review by abhijith-biju — COMMENTED** [review]
Please look into the comments.

**Review by abhijith-biju — APPROVED** [review]

**Review by RJR-003 — APPROVED** [review]

**Pull request merged**